.github/workflows/integration.yaml

```diff
@@ -41,18 +41,31 @@ jobs:
           username: "${{ secrets.IMAGE_REPO_USERNAME }}"
           password: "${{ secrets.IMAGE_REPO_PASSWORD }}"
 
-      - name: Build Image
+      - name: Generate Currency Converter Dockerfile
+        working-directory: tutorials/currency-converter
+        run: mvn -B docker:build -Ddocker.buildArchiveOnly # Skip building image, just create Dockerfile
+
+      - name: Build Data Generator Image
         if: github.event_name != 'push'
         uses: docker/build-push-action@v6
         with:
           context: tutorials/data-generator/
           platforms: linux/amd64,linux/arm64
           push: false
           file: tutorials/data-generator/Dockerfile
+
+      - name: Build Currency Converter Image
```

Comment:
It is not needed for this PR, but we could move all the docker build and push steps to another workflow (which depends on this build workflow) and only triggers when certain [paths](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore) are changed. i.e. only those that alter the content of the containers.